### PR TITLE
FLUID-5686: Collapsed emphasize links and inputs larger into enhance links

### PR DESCRIPTION
**Pull request by jobara**
The new enhancedLinks preference controls enhancing both links and inputs. Also, uses an on/off toggle instead of a straight checkbox.

https://issues.fluidproject.org/browse/FLUID-5686

Related Docs PR: https://github.com/fluid-project/infusion-docs/pull/117

Requires https://github.com/fluid-project/infusion/pull/811 to go in first.

### CODE REVIEW

**Comment by cindyli** [comment]
Please fix linting errors.

File: src/framework/preferences/html/PrefsEditorTemplate-enhanceInputs.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<h2><span class="fl-icon-links"></span><span class="flc-prefsEditor-label fl-heading-text"></span></h2>
```
Comment (cindyli) [inline src/framework/preferences/html/PrefsEditorTemplate-enhanceInputs.html]: Shall `fl-icon-links` be `fl-icon-inputs`?

File: tests/framework-tests/preferences/js/EnactorsTests.js [hunk 2]
```diff
@@ -81,87 +81,47 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
...
 
     /*******************************************************************************
-     * Unit tests for fluid.prefs.enactor.emphasizeLinks
+     * Unit tests for fluid.prefs.enactor.enhanceInputs
      *******************************************************************************/
 
-    fluid.defaults("fluid.tests.emphasizeLinksTests", {
+    fluid.defaults("fluid.tests.enhanceInputsTests", {
         gradeNames: ["fluid.test.testEnvironment"],
         container: ".flc-emphasizeLinks",
```
Comment (cindyli) [inline tests/framework-tests/preferences/js/EnactorsTests.js]: The container selector should be `.flc-enhanceInputs`.

Comment (cindyli) [inline tests/framework-tests/preferences/js/EnactorsTests.js]: This line doesn't seem needed. Please double check if it can be removed.

**Comment by jobara** [comment]
@cindyli ready for more review.

File: src/framework/preferences/css/stylus/Enactors.styl [hunk 3]
```diff
@@ -26,14 +26,14 @@
 }
 
 // UI Enhancer "Enlarge text" input/button options
-.fl-text-larger input,
-.fl-text-larger button,
-.fl-text-larger .fl-input-enhance {
+.fl-input-enhanced input,
```
Comment (cindyli) [inline src/framework/preferences/css/stylus/Enactors.styl]: Shall `textarea` tag be considered as a type of input and be enhanced too?

Comment (jobara) [inline src/framework/preferences/css/stylus/Enactors.styl]: Perhaps we should do this for [form elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element#Forms) in general.

Comment (jobara) [inline src/framework/preferences/css/stylus/Enactors.styl]: We talked more about this in the [IRC Channel](https://botbot.me/freenode/fluid-work/2017-04-12/?msg=83924100&page=2). The conclusion was to just add styles for `<textarea>` and `<select>`.

**Comment by jobara** [comment]
@cindyli this is ready for more review.

**Comment by cindyli** [comment]
[The distribute options for `linksControls` panel in PrefsEditor.js](https://github.com/jobara/infusion/blob/FLUID-5686/src/framework/preferences/js/PrefsEditor.js#L142-L144) should be changed to `enhanceInputs`.

**Comment by cindyli** [comment]
These tests are still testing `links` preference:
tests/framework-tests/preferences/js/FullPreviewPrefsEditorTests.js, line 37
tests/framework-tests/preferences/js/UIEnhancerTests.js, line 147

Should they be replaced with tests for `enhance inputs`?

**Comment by jobara** [comment]
@cindyli regarding the tests, I believe those are tests for composite panels, for which the old emphasizeLinks was the only one. The tests should have all of the structure needed for running the test, so I think it's okay.

**Comment by cindyli** [comment]
@jobara regarding the tests, it seems they are not for testing composite panels, instead, `links` options are sitting there and not be tested or used at all. Can you double check if they can be removed? Thanks.

**Comment by jobara** [comment]
@cindyli sorry, was looking at the wrong files,  ready for more review.

**Comment by cindyli** [comment]
Merged at 0ca2eac0a06731f99588399c9abd3e9d29732235

**Pull request merged**